sstables/mx/partition_reversing_data_source.cc

```diff
@@ -0,0 +1,608 @@
+    }
+    data_consumer::processing_result process_state(temporary_buffer<char>& data) {
+        _processing_data = &data;
+        auto ret = _gen.generate();
+        return ret;
+    }
+private:
+    uint64_t current_position() {
+        return position() - _processing_data->size();
+    }
+    processing_result_generator do_process_state() {
+        while (true) {
+            _finished = false;
+            co_yield read_8(*_processing_data);
+            auto flags = unfiltered_flags_m(_u8);
+            _current_tombstone_reversing_info.reset();
+            if (flags.is_end_of_partition()) {
+                _end_of_partition = true;
+                _finished = true;
+                co_yield data_consumer::proceed::no;
+                break;
+            } else if (flags.is_range_tombstone()) {
+                _current_tombstone_reversing_info.emplace();
+                _current_tombstone_reversing_info->kind_offset = current_position();
+                co_yield read_8(*_processing_data);
+                _current_tombstone_reversing_info->range_tombstone_kind = bound_kind_m(_u8);
+                co_yield read_16(*_processing_data);
+                _ck_size = _u16;
+                if (_ck_size != 0) {
+                    _reading_range_tombstone_ck = true;
+                }
+            } else {
+                if (flags.has_extended_flags()) {
+                    // skip flags
+                    co_yield read_8(*_processing_data);
+                    // `row_body_skipping_context` should not be constructed on static rows
+                    auto extended_flags = unfiltered_extended_flags_m(_u8);
+                    assert(!extended_flags.is_static());
+                }
+                _ck_size = _column_translation.clustering_column_value_fix_legths().size();
+                _reading_partition_ck = true;
+            }
+            if (_reading_partition_ck || _reading_range_tombstone_ck) {
+                setup_ck(_column_translation.clustering_column_value_fix_legths());
+                while (!no_more_ck_blocks()) {
+                    if (should_read_block_header()) {
+                        co_yield read_unsigned_vint(*_processing_data);
+                        _ck_blocks_header = _u64;
+                    }
+                    if (is_block_null() || is_block_empty()) {
+                        move_to_next_ck_block();
+                        continue;
+                    }
+                    // possibly read the length of, and then skip the clustering cell
+                    if (auto len = get_ck_block_value_length()) {
+                        co_yield skip(*_processing_data, *len);
+                    } else {
+                        co_yield read_unsigned_vint(*_processing_data);
+                        co_yield skip(*_processing_data, _u64);
+                    }
+                    move_to_next_ck_block();
+                }
+                _reading_partition_ck = false;
```

Comment:
"partition ck" is a bit confusing. Partitions don't have clustering keys any more than they have range tombstone keys. They have rows, and rows have clustering keys.